src/herder/TransactionQueue.cpp

```diff
@@ -292,24 +292,27 @@ TransactionQueue::sourceAccountPending(AccountID const& accountID) const
 bool
 validateSorobanMemo(TransactionFrameBasePtr tx)
 {
-    if (tx->getEnvelope().type() != ENVELOPE_TYPE_TX)
+    // v0 envelopes are invalid from protocol version 13 onwards, so it's
```

Comment:
This is still kind of brittle. Would it be hard to move the method to TransactionFrameBase with proper overrides, instead of looking directly at the XDR in the queue?

Reply:
I pushed an update. I'm not really a fan of this method living in `TransactionFrameBase`, but it is less error prone now.